test/files/neg/t2102.check

```diff
@@ -1,6 +1,6 @@
 t2102.scala:2: error: type mismatch;
- found   : java.util.Iterator[Int]
- required: scala.collection.Iterator[_]
+ found   : Iterator[Int] (in java.util)
+ required: Iterator[_] (in scala.collection)
```

Comment:
I think that's less readable. Most of the point of this PR is to bring the identical parts of the name all the way to the left, so that it's easily recognizable what the problem is. This way, the identical parts of the name are pushed a varying amount to the right.